src/Partials/InputField.vala

```diff
@@ -89,138 +93,41 @@ public class Akira.Partials.InputField : Gtk.EventBox {
-
-        button_up.enter_notify_event.connect (event => {
-            button_up_image.opacity = 1;
-            return false;
-        });
-        button_up.leave_notify_event.connect (event => {
-            button_up_image.opacity = 0.6;
-            return false;
-        });
-
-        var button_down_image = new Gtk.Image.from_icon_name ("pan-down-symbolic", Gtk.IconSize.MENU);
-        button_down_image.get_style_context ().add_class ("input-button");
-        button_down_image.get_style_context ().add_class ("down");
-        button_down_image.opacity = 0.6;
-
-        button_down = new Gtk.EventBox ();
-        button_down.add (button_down_image);
-        button_down.event.connect (button_down_event);
-
-        button_down.enter_notify_event.connect (event => {
-            button_down_image.opacity = 1;
-            return false;
-        });
-        button_down.leave_notify_event.connect (event => {
-            button_down_image.opacity = 0.6;
-            return false;
-        });
-
-        spin_grid.attach (button_up, 0, 0, 1, 1);
-        spin_grid.attach (button_down, 0, 1, 1, 1);
-
-        overlay.add (entry);
-        overlay.add_overlay (spin_grid);
-        add (overlay);
-
-        enter_notify_event.connect (event => {
-            if (!entry.sensitive || entry.has_focus) {
-                return false;
-            }
-
-            if (event.detail != Gdk.NotifyType.INFERIOR) {
-                spin_grid.visible = true;
-                spin_grid.no_show_all = false;
-                spin_grid.show_all ();
-                if (icon_right) {
-                    entry.secondary_icon_name = "";
-                } else {
-                    entry.primary_icon_name = "";
-                }
-            }
-            return false;
-        });
-
-        leave_notify_event.connect (event => {
-            if (event.detail != Gdk.NotifyType.INFERIOR) {
-                spin_grid.visible = false;
-                spin_grid.no_show_all = true;
-                if (icon_right) {
-                    entry.secondary_icon_name = icon;
-                } else {
-                    entry.primary_icon_name = icon;
-                }
-            }
-            return false;
-        });
-
-        entry.focus_in_event.connect (() => {
-            spin_grid.visible = false;
-            spin_grid.no_show_all = true;
-            if (icon_right) {
-                entry.secondary_icon_name = icon;
-            } else {
-                entry.primary_icon_name = icon;
-            }
-        });
+        add (entry);
+    }
+
+    public void set_range (double min_value, double max_value) {
+        entry.set_range (min_value, max_value);
     }
 
-    private bool handle_key (Gdk.EventKey key) {
+    private bool handle_key_press (Gdk.EventKey key) {
         // Arrow UP
-        if (key.keyval == 65362) {
-            increase_value (key);
+        if (key.keyval == 65362 && key.state == Gdk.ModifierType.SHIFT_MASK) {
```

Comment:
Here, from a mere code readability point of view, it's better to use the `Gdk.Key.uparrow` and `Gdk.Key.downarrow` enum values, so it's immediately clear which key we are referring to.